src/iam/jwt.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (c) 2018, Novo Nordisk Foundation Center for Biosustainability,
+# Technical University of Denmark.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Handling and verification of JWT claims."""
+
+import logging
+from functools import wraps
+
+from flask import abort, g, request
+from jose import jwt
+
+
+logger = logging.getLogger(__name__)
+
+
+def init_app(app):
+    """Add the jwt decoding middleware to the app."""
+    @app.before_request
+    def decode_jwt():
+        if 'Authorization' not in request.headers:
+            logger.debug("No JWT provided")
+            g.jwt_valid = False
+            g.jwt_claims = {'prj': {}}
+            return
+
+        auth = request.headers['Authorization']
+        if not auth.startswith('Bearer '):
+            g.jwt_valid = False
+            g.jwt_claims = {'prj': {}}
+            return
+
+        try:
+            _, token = auth.split(' ', 1)
```

Comment:
```suggestion
            token = auth.split(' ', 1)[1]
```
Could also use `auth.split(maxsplit=1)` for splitting on any white space.